File: src/main/java/com/university/view/pdf/PrepaymentReportPdfReport.java
package com.university.view.pdf;

import com.itextpdf.text.*;
import com.itextpdf.text.pdf.PdfPCell;
import com.itextpdf.text.pdf.PdfPTable;
import com.itextpdf.text.pdf.PdfWriter;
import com.university.entities.PrepaymentReport;
import com.university.entities.User;

import java.io.ByteArrayInputStream;
import java.io.ByteArrayOutputStream;
import java.util.List;
import java.util.logging.Level;
import java.util.logging.Logger;

public class PrepaymentReportPdfReport implements PdfReport<PrepaymentReport> {
    @Override
    public ByteArrayInputStream getReport(List<PrepaymentReport> prepaymentReports) {

        Document document = new Document();
        ByteArrayOutputStream out = new ByteArrayOutputStream();

        try {

            PdfPTable table = new PdfPTable(7);
            table.setWidthPercentage(100);
            table.setWidths(new int[]{2, 5, 5, 5, 5, 5, 5});

            Font headFont = FontFactory.getFont(FontFactory.TIMES_ROMAN);

            PdfPCell headCell;

            headCell = new PdfPCell(new Phrase("#", headFont));
            headCell.setHorizontalAlignment(Element.ALIGN_CENTER);
            table.addCell(headCell);

            headCell = new PdfPCell(new Phrase("Fare", headFont));
            headCell.setHorizontalAlignment(Element.ALIGN_CENTER);
            table.addCell(headCell);

            headCell = new PdfPCell(new Phrase("Seat reservation", headFont));
            headCell.setHorizontalAlignment(Element.ALIGN_CENTER);
            table.addCell(headCell);

            headCell = new PdfPCell(new Phrase("Hotel accommodation", headFont));
            headCell.setHorizontalAlignment(Element.ALIGN_CENTER);
            table.addCell(headCell);

            headCell = new PdfPCell(new Phrase("Telephone conversations", headFont));
            headCell.setHorizontalAlignment(Element.ALIGN_CENTER);
            table.addCell(headCell);

            headCell = new PdfPCell(new Phrase("Sum per days", headFont));
            headCell.setHorizontalAlignment(Element.ALIGN_CENTER);
            table.addCell(headCell);

            headCell = new PdfPCell(new Phrase("Preparation date", headFont));
            headCell.setHorizontalAlignment(Element.ALIGN_CENTER);
            table.addCell(headCell);


            for (PrepaymentReport report: prepaymentReports) {

                PdfPCell cell;

                cell = new PdfPCell(new Phrase(String.valueOf(report.getId())));
                cell.setPaddingLeft(5);
                cell.setHorizontalAlignment(Element.ALIGN_RIGHT);
                table.addCell(cell);

                cell = new PdfPCell(new Phrase(String.valueOf(report.getFare())));
                cell.setPaddingLeft(5);
                cell.setHorizontalAlignment(Element.ALIGN_LEFT);
                table.addCell(cell);

                cell = new PdfPCell(new Phrase(String.valueOf(report.getSeatReservation())));
                cell.setPaddingLeft(5);
                cell.setHorizontalAlignment(Element.ALIGN_LEFT);
                table.addCell(cell);

                cell = new PdfPCell(new Phrase(String.valueOf(report.getHotelAccommodation())));
                cell.setPaddingLeft(5);
                cell.setHorizontalAlignment(Element.ALIGN_LEFT);
                table.addCell(cell);

                cell = new PdfPCell(new Phrase(String.valueOf(report.getTelephoneConversations())));
                cell.setPaddingLeft(5);
                cell.setHorizontalAlignment(Element.ALIGN_LEFT);
                table.addCell(cell);

                cell = new PdfPCell(new Phrase(String.valueOf(report.getSumPerDiems())));
                cell.setPaddingLeft(5);
                cell.setHorizontalAlignment(Element.ALIGN_LEFT);
                table.addCell(cell);

                cell = new PdfPCell(new Phrase(String.valueOf(report.getPreparationDate())));
                cell.setPaddingLeft(5);
                cell.setHorizontalAlignment(Element.ALIGN_LEFT);
                table.addCell(cell);
            }

            PdfWriter.getInstance(document, out);
            document.open();
            document.add(table);
            document.close();

        } catch (DocumentException ex) {
            Logger.getLogger(UserPdfReport.class.getName()).log(Level.SEVERE, null, ex);
        }

        return new ByteArrayInputStream(out.toByteArray());
    }
}

File: src/main/kotlin/com/university/services/pdf/PrepaymentReportPdfReport.kt
package com.university.services.pdf

import com.itextpdf.text.Document
import com.itextpdf.text.Element.ALIGN_LEFT
import com.itextpdf.text.Element.ALIGN_RIGHT
import com.itextpdf.text.FontFactory
import com.itextpdf.text.pdf.PdfPTable
import com.itextpdf.text.pdf.PdfWriter
import com.university.entities.PrepaymentReport
import java.io.ByteArrayInputStream
import java.io.ByteArrayOutputStream

class PrepaymentReportPdfReport : PdfReport<PrepaymentReport> {
    override fun getReport(list: List<PrepaymentReport>): ByteArrayInputStream {
        val document = Document()
        val out = ByteArrayOutputStream()
        val table = PdfPTable(7)
        table.widthPercentage = 100f
        table.setWidths(intArrayOf(2, 5, 5, 5, 5, 5, 5))
        val headFont = FontFactory.getFont(FontFactory.TIMES_ROMAN)
        table.addCell(createCell("#", headFont))
        table.addCell(createCell("Fare", headFont))
        table.addCell(createCell("Seat reservation", headFont))
        table.addCell(createCell("Hotel accommodation", headFont))
        table.addCell(createCell("Telephone conversations", headFont))
        table.addCell(createCell("Sum per days", headFont))
        table.addCell(createCell("Preparation date", headFont))
        for (report in list) {
            with(report) {
                table.addCell(createCell(id.toString(), ALIGN_RIGHT))
                table.addCell(createCell(fare.toString(), ALIGN_LEFT))
                table.addCell(createCell(seatReservation.toString(), ALIGN_LEFT))
                table.addCell(createCell(hotelAccommodation.toString(), ALIGN_LEFT))
                table.addCell(createCell(telephoneConversations.toString(), ALIGN_LEFT))
                table.addCell(createCell(sumPerDiems.toString(), ALIGN_LEFT))
                table.addCell(createCell(preparationDate.toString(), ALIGN_LEFT))
            }
        }
        PdfWriter.getInstance(document, out)
        document.open()
        document.add(table)
        document.close()
        return ByteArrayInputStream(out.toByteArray())
    }
}
